default.nix

```diff
@@ -37,19 +37,32 @@ let
         livereload
       ]);
       script = ''
-        from livereload import Server, shell
+        from livereload import Server
+        from subprocess import Popen, PIPE
+        import shlex
 
         server = Server()
 
-        build_docs = shell("make html")
+        def build():
+          p = Popen(
+            shlex.split("nix-build -A build"),
```

Comment:
You're always doing a `nix-build` now, so it can't reuse any previously built artifacts. What's the difference in responsiveness on incremental changes?

Reply:
Yes, but it turns out not to matter. On my machine it's fast enough to be unnoticeable. Takes about as long as switching between editor and browser at a relaxed pace. At worst, you'll see the page contents updating.

Reply:
Running it myself, a `make html` runs in less than a second, whereas a `nix-build -A build` runs in 10 seconds..

Reply:
The number of times I had to restart the shell or `make clean` because something didn't get updated warrant the order of magnitude increase, for me at least. An out-of-tree `make html` would of course be better but also more work. This is correct but slow, now.

Reply:
> `make clean` because something didn't get updated

Then that's a `Makefile` issue we should fix. I really don't think we should use Nix for incremental development, it's not made for that

Reply:
Discussed in the meeting, 10 seconds is a bit long but still fine, and the sphinx problem is annoying. If it gets much longer in the future this needs to be reconsidered.